torch_audiomentations/core/composition.py

```diff
@@ -0,0 +1,62 @@
+import random
+from typing import List
+
+import torch
+
+
+class Compose(torch.nn.Module):
+    """This class can apply a sequence of transforms to waveforms."""
+
+    def __init__(
+        self,
+        transforms: List[torch.nn.Module],
+        shuffle: bool = False,
+        p: float = 1.0,
+        p_mode="per_batch",
+    ):
+        """
+
+        :param transforms: List of waveform transform instances
+        :param shuffle: Should the order of transforms be shuffled?
+        :param p: The probability of applying the Compose to the given batch.
+        :param p_mode: Only "per_batch" is supported at the moment.
+        """
+        super().__init__()
+        self.transforms = transforms
+        self.p = p
+        if p_mode != "per_batch":
+            # TODO: Support per_example as well? And per_channel?
+            raise Exception('p_mode = "{}" is not supported')
+        self.p_mode = p_mode
+        self.shuffle = shuffle
+
+        name_list = []
+        for transform in self.transforms:
+            name_list.append(type(transform).__name__)
+        self.__name__ = "_".join(name_list)
```

Comment:
Should `__name__` somehow reflect the state of `shuffle`?
Maybe `__name__` is not that important anyway...

For instance, though using emojis is probably not a good idea:
* `Gain ⏩ PolarityInversion` when `shuffle` is `False` 
* `Gain 🔀 PolarityInversion` when `shuffle` is `True`

Reply:
Yeah, I don't know if this is important. It's copy-pasta from audiomentations 🙈 Should I remove it?

Reply:
Depends on whether you had any use for this `__name__` thing in mind?

Reply:
Nope. I think I'll remove it for now